src/bitcoind/interface.rs

```diff
@@ -628,17 +628,24 @@ impl BitcoinD {
         let utxos = self.list_since_block(db_tip)?;
         for utxo in utxos {
             if utxo.category == "receive" && utxo.label == Some(DEPOSIT_UTXOS_LABEL.to_string()) {
-                // If I've never heard about it, and it's not already spent, then it's new
                 if deposits_utxos.get(&utxo.outpoint).is_none() {
-                    new_unconf.insert(
-                        utxo.outpoint,
-                        UtxoInfo {
-                            txo: utxo.txo,
-                            // All new utxos are marked as unconfirmed. This allows for a proper state
-                            // transition.
-                            is_confirmed: false,
-                        },
-                    );
+                    // Alright, I don't have this outpoint in the cache... But maybe I don't have
+                    // it because it's spent already! Let's check.
```

Comment:
What is a scenario where this check is needed?

Reply:
Let's keep in mind that `sync_deposits` could be run multiple times with the same `previous_tip`.
If between the `previous_tip` and the `current_tip` we receive and spend a deposit (let's call it D), at the first call of `sync_deposits` we will mark D as unconfirmed, at the second one as confirmed, at the third one as spent (the poller will proceed to remove it from the deposits cache), at the fourth one unconfirmed again.
You can reproduce a situation similar to this one removing the "already spent check" and running `test_reorged_deposit_status` - I believe what's happening in the test is slightly different, due to all the reorgs etc, but the main idea behind the bug should be the same

Reply:
I think this prevents syncing the history you missed (https://github.com/revault/revaultd/pull/352#issuecomment-1013754522). And that's the entire point of refactoring this code to use `listsinceblock`..

You are right that we are missing information here, Checking whether the new deposit does not correspond to a known spent vault should probably be done by the caller, that's bulky but probably fine as it would be a very edgy case (that i still fail to *precisely* model) that would be resolved by a new block.

> If between the previous_tip and the current_tip we receive and spend a deposit (let's call it D), at the first call of sync_deposits we will mark D as unconfirmed, at the second one as confirmed, at the third one as spent (the poller will here proceed to remove D from the deposits cache), at the fourth one unconfirmed again.

So we need to poll 4 times with the same `previous_tip` and `current_tip`? Therefore it must be the latest block. So you'd need a deposit to become confirmed, unvaulted and spent in the same block. That's not possible. Is there an edge case where this can happen? Maybe a deep reorg where the deposit shifts many blocks back?
Do you have a concrete example where this happens?